apps/language_server/test/experimental/code_mod/format_test.exs

```diff
@@ -71,6 +71,76 @@ defmodule ElixirLS.Experimental.FormatTest do
       ]t == result
     end
 
+    test "it can split a long line to two lines" do
+      unformatted = ~q[
+        defmodule Unformatted do
+          def foo1(s) do
+            s = very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) |> IO.inputs()
+            s
+          end
+
+          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) do
+            s
+          end
+        end
+      ]t
+
+      formatted = ~q[
+        defmodule Unformatted do
+          def foo1(s) do
+            s =
+              very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) |> IO.inputs()
+
+            s
+          end
+
+          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) do
+            s
+          end
+        end
+      ]t
+
+      assert {:ok, formatted} == modify(unformatted)
+    end
```

Comment:
this looks like two different tests, can they be split apart?

Reply:
Hi, Steve, thanks for your suggestion.

I just used it as an input-output comparison. I don't know if you are referring to splitting the test like this:

```elixir
    test "it can split a long line to two lines" do
      unformatted = ~q[
        defmodule Unformatted do
          def foo1(s) do
            s = very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) |> IO.inputs()
            s
          end

          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) do
            s
          end
        end
      ]t

      expected = ~q[
        defmodule Unformatted do
          def foo1(s) do
            s =
              very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) |> IO.inputs()

            s
          end

          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonooooong(s) do
            s
          end
        end
      ]t

      assert {:ok, result} = modify(unformatted)
      assert result == expected
    end
```

If so, I would like to hear from you about the benefits of doing so.

Reply:
for the long lines test, you defined a function and a call to that function, I'm assuming that there were formatting issues on both of them, they should be defined in two separate tests, one that tests the formatting of a function definition and the other that tests the formatting of a function call. This way, an error in one of the tests indicates a more specific problem than a test that covers both cases.

Reply:
I see what you mean, but I've tried, and It's hard for me to finding an example of a long function definition being formatted as **two** lines(only function call will). Usually it's only formatted as **three** lines, like the code below, and I've added this example in the last commit.

```elixir
      unformatted = ~q[
        defmodule Unformatted do
          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong(s) do
            s
          end
        end
      ]t

      formatted = ~q[
        defmodule Unformatted do
          def very_loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong(
                s
              ) do
            s
          end
        end
      ]t

      assert {:ok, formatted} == modify(unformatted)
```